### PR TITLE
cell.csl: Add a macro for DOI and webpage accessed date.

### PR DESCRIPTION
**Pull request by nvall**
Hi,
- I recently found that Cell journals require the use of DOI at the end of the reference list see: https://www.cell.com/cell/authors#format `We encourage the inclusion of DOIs in all journal article references`
- Some references are sometimes from webpages and it is better to add the accessed date.

This pull request adds a macro to display DOI for articles and Accessed date to webpages.

Hope this will be usefull,
Best,
Nicolas

### CODE REVIEW

**Comment by bwiernik** [comment]
Thanks! Do you have an actual example of a website citation in the journal having access dates? Most citation styles do _not_ include access dates for website citations. So unless there is information otherwise for Cell, please remove that.

File: cell.csl [hunk 1]
```diff
@@ -57,6 +57,36 @@
...
     </names>
   </macro>
+  <macro name="accessed-doi">
+    <choose>
+      <if type="article-newspaper" match="none">
+        <choose>
+          <if variable="DOI">
+            <text value=" doi:"/>
+            <text variable="DOI"/>
+          </if>
```
Comment (bwiernik) [inline cell.csl:67]:
```suggestion
          <if variable="DOI">
            <text variable="DOI" prefix="https://doi.org/"/>
          </if>
```

This is the preferred format from CrossRef

**Review by bwiernik — CHANGES_REQUESTED** [review]

**Comment by nvall** [comment]
Thanks for your suggestions. I have remove URL part of the changes and added a bit of formating so it fits with Cell guidelines.

**Review by nvall — COMMENTED** [review]
Thanks, I pushed a new version with DOI corrected and URL removed.

File: cell.csl [hunk 2]
```diff
@@ -57,6 +57,13 @@
       <label form="short" prefix=", "/>
     </names>
   </macro>
+  <macro name="doi">
+    <choose>
+      <if variable="DOI">
+        <text variable="DOI" prefix="https://doi.org/"/>
+      </if>
+    </choose>
```
Comment (bwiernik) [inline cell.csl:65]:
```suggestion
    <choose>
      <if variable="DOI">
        <text variable="DOI" prefix="https://doi.org/"/>
      </if>
      <else-if type="webpage post post-weblog" match="any">
        <text variable="URL"/>
      </else-if>
    </choose>
```

**Review by bwiernik — CHANGES_REQUESTED** [review]
I think it would be good to include the URL for webpages if they don't have a DOI. Otherwise there is basically no retrieval information at all

**Comment by bwiernik** [comment]
Thanks!

**Pull request merged**